src/utils/chunks/upload.rs

```diff
@@ -13,6 +13,10 @@ use crate::utils::progress::ProgressBar;
 /// A trait representing options for chunk uploads.
 /// The trait also provides the `upload_chunked_objects` method, which
 /// executes the upload given the options and a list of chunked objects.
+/// New code should use `GenericChunkOptions` where possible, instead of
+/// creating custom implementations of this trait.
+/// The trait exists for backwards compatibility with older code, in
+/// particular by allowing DifUpload to be used as ChunkOptions.
```

Comment:
In light of this, is the plan to remove the trait eventually?

Reply:
Yeah, I think it would make sense eventually to just have the `GenericChunkOptions` struct. Although I am not sure, maybe having the trait would make this code more flexible. 

What do you think @loewenheim? Do you see any elegant way to go with the struct already from the beginning?

Reply:
I suppose we could replace the new `ChunkedDifUpload` struct from #2302 with a method on `DifUpload`, which given the `ChunkServerOptions` would create a new `GenericChunkOptions` struct.

Then, we could have the `GenericChunkOptions` struct be renamed to `ChunkOptions`, replacing the `ChunkOptions` trait

Reply:
That could be a good idea, but it doesn't have to be done right away.

Reply:
@loewenheim I am already working on it. I think it will be a clearer solution this way.